app/controllers/userController.js

```diff
@@ -1,13 +1,15 @@
+import httpStatus from 'http-status'
 import UserService from '../services/userService'
+import responseWrapper from '../../lib/helpers/response-wrapper'
 
 class UserController {
   static async signup(req, res, next) {
-    try {
-      await UserService.createUser(req.body)
-      return res.status(201).json({ message: 'signup successful' })
-    } catch (error) {
-      return next(error)
-    }
+    await UserService.createUser(req.body)
+      .then(user => {
```

Comment:
you do not have to explicitly resolve the promise in the `.then()` as this is what `await` does. When you have `await`,  the execution will wait for the process on that line to complete before moving on to the next one.